### PR TITLE
Add password authentication for SMTP server

### LINKED ISSUE #757: Email server settings
I am looking to configure IRIDA to use an existing SMTP server to send mail. This server requires password authentication but the IRIDA config does not expose this option. Perhaps I am not understanding a limitation of the underlying framework.
There isn't any documentation that I can find surrounding the mail server configuration beyond the mention in the web.conf template.

Is it possible to add this feature?

## Additional information
https://github.com/phac-nml/irida/blob/a21c59fe30b80a3dda8fb1967463bbc4e45e2d6d/src/main/java/ca/corefacility/bioinformatics/irida/config/services/WebEmailConfig.java#L51-L57

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/mail/javamail/JavaMailSenderImpl.html#setPassword-java.lang.String-

### PR DESCRIPTION
**Pull request by innovate-invent**
## Description of changes
Add password and port parameter for smtp user authentication
web.conf can now include:
```
mail.server.password=
mail.server.port=
```

Testing this can be done by configuring IRIDA to use a SMTP server with password authentication and requesting a password reset from the IRIDA web interface.

## Related issue
Fixes #757 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [ ] User documentation updated for UI or technical changes.

### CODE REVIEW

**Comment by innovate-invent** [comment]
I am not sure how tests for this should be done as it would require an smtp server.
I may also need to expose more smtp parameters to configure the different TLS settings.

**Review by tom114 — CHANGES_REQUESTED** [review]
Thanks for the PR @innovate-invent.  Since this is adding more customization options for connecting to a mail server, I asked below for a small refactor for how the default config values are set & read.  Since some may be optional settings now, it would be a good idea to add some `null` & empty checks for them before setting them on the mail sender.

Feel free to give me a shout if you have any questions about the requested changes.

File: src/main/resources/configuration.properties [hunk 1]
```diff
@@ -7,6 +7,8 @@ mail.server.host=YOUR_MAIL_HOST
 mail.server.protocol=YOUR_MAIL_PROTOCOL (usually SMTP)
 mail.server.email=YOUR_MAIL_SOURCE (the address the mail is coming from)
 mail.server.username=YOUR_MAIL_NAME (the name that the e-mail is coming from)
+mail.server.password=YOUR_MAIL_PASSWORD (the password associated with the provided name)
+mail.server.port=25
```
[the remark this answers is not in the document]
Comment (innovate-invent) [reply src/main/resources/configuration.properties]: done

File: src/main/java/ca/corefacility/bioinformatics/irida/config/services/WebEmailConfig.java [hunk 2]
```diff
@@ -50,9 +59,19 @@ public static PropertySourcesPlaceholderConfigurer propertySourcesPlaceholderCon
...
 	public ConfigurableJavaMailSender javaMailSender() {
 		ConfigurableJavaMailSenderImpl sender = new ConfigurableJavaMailSenderImpl();
+		Properties props = new Properties();
+
 		sender.setHost(host);
+		sender.setPort(port);
 		sender.setProtocol(protocol);
 		sender.setUsername(username);
+
+		if (!(UNCONFIGURED_PASSWORD_VALUE.equals(password) || password.isBlank())) {
```
[the remark this answers is not in the document]
Comment (innovate-invent) [reply src/main/java/ca/corefacility/bioinformatics/irida/config/services/WebEmailConfig.java]: port doesn't require a check as Java sets its default value

**Review by tom114 — CHANGES_REQUESTED** [review]
Thanks @innovate-invent , this change should be good to go!  Can you clear up the conflict in the changelog and it should be ready to merge.

**Review by tom114 — APPROVED** [review]

**Pull request merged**